site_scons/components/__init__.py

```diff
@@ -172,7 +172,19 @@ def define_mercury(reqs):
         MERCURY_DEBUG = '-DMERCURY_ENABLE_DEBUG=OFF '
     retriever = \
         GitRepoRetriever('https://github.com/mercury-hpc/mercury.git',
-                         True)
+	True)
+
+    reqs.define("UCX", libs=['ucp'])
+
+    if reqs.check_component('UCX'):
+        UCX = '-DNA_USE_UCX=ON ' \
+        '-DUCX_INCLUDE_DIR=/usr/include '\
+        '-DUCP_LIBRARY=/usr/lib64/libucp.so '\
+        '-DUCS_LIBRARY=/usr/lib64/libucs.so '\
+        '-DUCT_LIBRARY=/usr/lib64/libuct.so '
```

Comment:
You should also append 'UCX' (or 'ucx' if you follow suggestion above) to libs list so it gets added to requires= line in the mercury definition.   It should be listed as a requirement of mercury if it's available.

Reply:
Add libs.append('ucx') so the dependence is explicit on line 210